src/orange/OrangeTrackView.hh

```diff
@@ -203,13 +204,31 @@ OrangeTrackView::operator=(const Initializer_t& init)
     local.temp_sense = this->make_temp_sense();
 
     // Initialize logical state
-    auto tracker = this->make_tracker(UniverseId{0});
-    auto tinit   = tracker.initialize(local);
-    // TODO: error correction/graceful failure if initialiation failured
-    CELER_ASSERT(tinit.volume && !tinit.surface);
+    size_type uid = 0;
+
+    detail::UnitIndexer unit_indexer(params_.unit_indexer_data);
+
+    while (true)
+    {
+        auto tracker = this->make_tracker(UniverseId{uid});
+        auto tinit   = tracker.initialize(local);
+        // TODO: error correction/graceful failure if initialiation failured
+        CELER_ASSERT(tinit.volume && !tinit.surface);
+
+        OpaqueId<VolumeRecord> vrid(tinit.volume.unchecked_get());
 
-    // Save local data
-    states_.vol[thread_] = tinit.volume;
+        auto daughter_id = params_.volume_records[vrid].daughter;
+
+        if (daughter_id)
+        {
+            uid = daughter_id.unchecked_get();
+            continue;
+        }
+
+        states_.vol[thread_]
+            = unit_indexer.global_volume(UniverseId{uid}, tinit.volume);
```

Comment:
We probably want to retain the volume ID at each level, but that can be deferred to another merge.